### PR TITLE
Add `GitHub.putIssueComment` helper

### PR DESCRIPTION
**Pull request by 72636c**

### CODE REVIEW

File: src/api/github/issueComment.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import { getPullRequest } from './pullRequest';
+
+const getUserId = async (client: Octokit): Promise<string> => {
+  const { data } = await client.users.getAuthenticated();
+
+  return data.login;
+};
+
+/**
+ * {@link https://docs.github.com/en/rest/reference/checks#create-a-check-run}
```
Comment (samchungy) [inline src/api/github/issueComment.ts:14]: link doesn't seem right fyi

File: src/api/github/issueComment.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  const dir = process.cwd();
+
+  const { owner, repo } = await Git.getOwnerAndRepo({ dir });
+
+  const client = new Octokit({ auth: apiTokenFromEnvironment() });
+
+  const issueNumber =
+    params.issueNumber ??
+    (await getPullRequest({ client, env }).then(({ number }) => number));
```
Comment (samchungy) [inline src/api/github/issueComment.ts:81]:
```suggestion
    (await getPullRequest({ client, env })).number;
```

File: src/api/github/pullRequest.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  if (!data.length) {
+    throw new Error(
+      `Commit ${commitId} is not associated with a GitHub pull request`,
+    );
+  }
+
+  data
+    .sort((a, b) => b.updated_at.localeCompare(a.updated_at))
+    .sort((a, b) => {
+      if (typeof a.closed_at === typeof b.closed_at) {
```
Comment (samchungy) [inline src/api/github/pullRequest.ts:71]: question: Does mean the bot would update a comment on a closed PR if there are no open PRs? Is it worth it to update a comment on a closed PR?

Comment (72636c) [reply src/api/github/pullRequest.ts:71]: Good catch, it probably makes sense to fail in this scenario.

File: src/api/github/pullRequest.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * Gets the number of the current pull request.
+ *
+ * This tries to extract the pull request from common CI environment variables,
+ * and falls back to querying the GitHub Repos API for the latest pull request
+ * associated with the head commit. An error is thrown if there are no
+ * associated pull requests or if they are all closed.
+ */
+export const getPullRequest = async (
```
Comment (samchungy) [inline src/api/github/pullRequest.ts:31]: question: Would it be worth exposing any of the other fields for the PR? I guess if I'm calling `getPullRequest` I would be expecting more than a simple number?

Comment (72636c) [reply src/api/github/pullRequest.ts:31]: Yeah good point. I wanted to keep the API small until there was a need for extra fields, but I figure we can add a few more non-controversial ones.

Comment (72636c) [reply src/api/github/pullRequest.ts:31]: On second thought, I'll scope this to `getPullRequestNumber`. The nicety here is that we can drive the function purely off of environment variables on the happy path.

**Review by samchungy — APPROVED** [review]

**Pull request merged**